api/src/main/java/org/pmiops/workbench/db/dao/EgressEventDao.java

```diff
@@ -11,6 +11,13 @@
 public interface EgressEventDao extends CrudRepository<DbEgressEvent, Long> {
   List<DbEgressEvent> findAllByUserAndStatusNot(DbUser user, EgressEventStatus status);
 
+  List<DbEgressEvent> findAllByUserAndWorkspaceAndCreationTimeBetweenAndEgressEventIdNot(
```

Comment:
We may want to  filter by status(`where status= REMEDIATED`). If all events are PENDING, we would still want to send at lease one email

Reply:
Why adding that filter would causing 0 emails send? Adding that filter will end up having less results in `prioEvents` list, so `shouldNotifyForEvent` is more likely to return true.

Initially I meant to add the status filter in the existing query. Then we would have creation time and also the status as filter. 
Then even two events are at the exact same millseconds, we would end up sending multiple alerts.
But if just having creation time filter, we would end up sending 0 emails.
So there is a decision between sending multiple emails vs 0 emails. 
But since you think this case is rare(as long as sumologic does not  sending duplicates event, we should be fine), then we may not need that extra filter

Reply:
Sorry you're right. That situation would send 2 emails and not 0. It would be way more common than two events having identical creation timestamps though.

> But if just having creation time filter, we would end up sending 0 emails.

This case will send 2 emails, because `BETWEEN` in MySQL is inclusive.